edb/edgeql/compiler/stmtctx.py

```diff
@@ -88,9 +89,33 @@ def init_context(
     env = context.Environment(
         schema=schema,
         options=options,
+        path_scope=(
+            inlining_context.path_scope
+            if inlining_context is not None else
+            None
+        ),
         alias_result_view_name=options.result_view_name,
+        scope_tree_nodes=(
+            inlining_context.env.scope_tree_nodes
+            if inlining_context is not None else
+            None
+        ),
+    )
+    ctx = context.ContextLevel(
+        None,
+        context.ContextSwitchMode.NEW,
+        env=env,
+        aliases=(
+            inlining_context.aliases
+            if inlining_context is not None else
+            None
+        ),
+        scope_id_ctr=(
+            inlining_context.scope_id_ctr
+            if inlining_context is not None else
+            None
+        ),
```

Comment:
We should use the existing context, with appropriate new scopes applied to it.
Doing it in a completely separate context is not going to work at all in anything but the most simple situations.

Reply:
And especially the same environment. (Or at least, an almost identical environment, since we might need to change options or something, though that is a little fraught too.)